install/install.sh

```diff
@@ -0,0 +1,389 @@
+        # keystone credentials such as OS_USERNAME must be set before.
+        python3 ./ministone.py token_issue  > /dev/null
+        if [ "$?" == "0" ]; then
+            return
+        fi
+        count=`expr ${count} \+ 1`
+        if [ ${count} -ge ${times} ]; then
+            echo -e "\033[1;31m ERROR: keystone didn't come up. Aborting... \033[m" 
+            exit 1
+        fi
+        sleep ${interval}
+    done
+}
+
+install(){
+    echo -e "\033[1;36m Begin installing Keystone... \033[m"
+    if [ ! "$(docker images | grep opensdsio/opensds-authchecker)" ]
+    then
+        echo -e "\033[3;33m Keystone image is not available locally. \033[m"
+        echo -e "\033[3;33m Pulling opensdsio/opensds-authchecker docker image... \033[m"
+        docker pull opensdsio/opensds-authchecker:latest
+    fi
+    echo -e "\033[3;33m Installing Keystone from docker image. \033[m"
+    echo -e "\033[3;33m Keystone docker image is pulled. Starting the container... \033[m"
+
+    docker run -d --privileged=true --restart=always --net=host --name=opensds-authchecker opensdsio/opensds-authchecker:latest > /dev/null
+
+    echo -e "\033[3;33m Copying the keystone configuration files... \033[m"
+    docker cp "./conf/keystone.policy.json" opensds-authchecker:/etc/keystone/policy.json
+
+    echo -e "\033[3;33m Setting the keystone credentials... \033[m"
+    keystone_credentials
+
+    echo -e "\033[3;33m Waiting for Keystone service to come up... \033[m"
+    wait_for_keystone
+    python3 ./ministone.py endpoint_bulk_update keystone "http://${HOST_IP}/identity"  > /dev/null
+
+    if [ ! "$(docker ps -a | grep opensds-authchecker)" ]
+    then
+        echo -e "\e[1;31m Something went wrong. Keystone is not installed. \e[0m" 
+    else
+        echo -e "\033[1;32m Keystone service is up and running. \033[m"
+        echo -e "\033[1;36m Begin installing SODA Dashboard... \033[m"
+        install_dashboard $@        
+        
+        if [[ $1 == 'install' ]]
+        then
+            if [ ! "$(docker ps -a | grep soda-dashboard)" ]
+            then
+                echo -e "\e[1;31m Something went wrong. SODA Dashboard is not installed. \e[0m" 
+            else
+                echo -e "\033[1;32m SODA Dashboard is installed successfully. \033[m"
+                echo -e "\033[1;32m You can login to the dashboard by typing http://$HOST_IP:8088 in the address bar. \033[m"
+            fi                    
+        fi        
+    fi
+}
+
+install_dashboard(){
+    
+    HOST_IP="${HOST_IP:=127.0.0.1}"
+    
+    echo -e "\033[3;33m The HOST IP is $HOST_IP \033[m"
+
+    echo -e "\033[3;33m The Dashboard Release version is $DASHBOARD_RELEASE_VERSION \033[m"
+
+    docker run -d --net=host --restart=always --name soda-dashboard \
+    -e OPENSDS_AUTH_URL=http://$HOST_IP/identity \
+    -e OPENSDS_HOTPOT_URL=http://$HOST_IP:50040 \
+    -e OPENSDS_GELATO_URL=http://$HOST_IP:8089 \
+    -e SODA_DELFIN_URL=http://$HOST_IP:8190 \
```

Comment:
Why these are hardcoded? The project selection is not available?! Or take such info from the soda conf information. Why this needs to hardcoded under dashboard? Should be part of SODA installer right?

Reply:
These are being exposed as environment variable to the dashboard docker container. Project has never been available for the Dashboard. 
There is no SODA conf information that is exported that gives these information on the env variables. These ports are internally mapped to the nginx conf within the SODA Dashboard docker container. This is the same command that was used as part of the SODA installer.

Reply:
I think this is not correct. We need to discuss and clean it. Currently can you please add FIXME and we can merge if it is a blocker.

Reply:
Sure we can do that. But this is how it works in the SODA installer currently.
We do not have any project / module specific configuration for Dashboard.

Reply:
Again, this is now an independent installer that needs to be run on the system without the common installer. In this case we will have to export all the required env variables on the command line and expose the same to the docker containers.